src/php/types/object.rs

```diff
@@ -161,6 +164,38 @@ impl ZendObject {
     pub(crate) fn refcount_inc(&mut self) {
         self.gc.refcount += 1;
     }
+
+    pub(crate) fn to_string(&self) -> String {
+        let mut ret = Zval::new();
+        unsafe {
+            zend_call_known_function(
+                (*self.ce).__tostring,
+                self as *const _ as *mut _,
+                self.ce,
+                &mut ret,
+                0,
+                ptr::null_mut(),
+                ptr::null_mut(),
+            );
+        }
+
+        if let Some(err) = ExecutorGlobals::take_exception() {
+            let class_name = unsafe { ZendString::from_ptr((*self.ce).name, false) };
```

Comment:
Passing `false` here feels weird. Would it maybe make sense to separate this into two type? A `ZendString` and `ZendStr` for example.

```rust
let class_name = unsafe { ZendStr::from_ptr((*self.ce).name) };

Reply:
Yeah I agree, I feel like I've got the FFI wrappers wrong (strings and arrays). What do you think about doing the same for strings? See [here](https://github.com/davidcole1340/ext-php-rs/blob/3646da741caf04849b3552915af34f68ee16eac7/src/php/types/array.rs) for progress with the array.

Implement traits on references to the underlying `zend_string`, and then have an `OwnedZendString` when you need to create one?

Reply:
I've changed the implementation of Zend strings, have a look [here](https://github.com/davidcole1340/ext-php-rs/commit/7ba9ee9bfb5d2888f6e83129aa4361cbb7cb43c5).

Reply:
Looks promising, but haven't had a chance to deep dive into it. Will hopefully be able to tonight, no promises.